uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature

```diff
@@ -0,0 +1,32 @@
+@GGAD
+Feature: GGMQ-1
+
+  As a developer, I can configure my client agents to connect and use MQTT
+
+  Background:
+    Given my device is registered as a Thing
+    And my device is running Greengrass
+    And my MQTT Client Control is running
+
+  Scenario: GGMQ-1-T1: As a customer, I can connect, subscribe and publish using client application to MQTT topic
+    When I create a Greengrass deployment with components
+      | aws.greengrass.clientdevices.Auth        | LATEST                                                        |
+      | aws.greengrass.clientdevices.mqtt.EMQX   | LATEST                                                        |
+      | aws.greengrass.clientdevices.IPDetector  | LATEST                                                        |
+      | aws.greengrass.client.Mqtt5JavaSdkClient | classpath:/greengrass/components/recipes/client_java_sdk.yaml |
+    Then the Greengrass deployment is COMPLETED on the device after 300 seconds
+    And agent "agent1" is connected to MQTT Client Control
+    When I create client device "test" with the following policy
+      | operation | resource     |
+      | connect   | "*"          |
+      | subscribe | "iot_data_0" |
+      | publish   | "iot_data_0" |
+    And I associate "test" with ggc
+    And I connect device "test" on "agent1" to "default_broker" as "agent1-test-default_broker"
+    Then connection "agent1-test-default_broker" is successfully established within 3 seconds
```

Comment:
Everything is very "agent" based here. Let's make this client-based as client is something well understood.

`Then client "test" is successfully connected within 3 seconds`

Make a reasonable agent name automatically and store the association between client and agent so that we don't need to keep referencing the agent here in the test scenario.

Reply:
We have a little more complicated architecture than in GGAT examples.
In GGAT we have a (maybe real) IoT device which is up once will infinite try to make connection with IoT or Greengrass device. In terms of GGAT client it is the same as MQTT connection. 
But it is not true for our architecture due to the fact that we have a client fully controlled. Initiatially when started it has no MQTT connection and just establishes bidirectional  gRPC link with Control. After that control can send requests to establish one or more MQTT connections. That means our 'client' is not equal to one MQTT connection. This way to avoid misunderstanding I offer a term agent. Which can have more than one MQTT connection. With that architecture we need to specify an agentId and connectionId pair to do connection-related operations like publish or subscribe. To simplify schema for end users I already implement logical connection names and how we can use any unique string to identify a particular connection despide of agent.